PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ['product.template']
-    _inherit = ["purchase.order"]
-    _inherit = ['purchase.order.line']
-    _inherit = ['sale.order']
-    _inherit = ['sale.order.line']

Total Changes: 10

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_purchase/models/product_template.py
+++ b/addons/sale_purchase/models/product_template.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class ProductTemplate(models.Model):
-    _inherit = ['product.template']
+    _inherit = 'product.template'
 
     service_to_purchase = fields.Boolean(
         "Subcontract Service", company_dependent=True, copy=False,

--- a/addons/sale_purchase/models/purchase_order.py
+++ b/addons/sale_purchase/models/purchase_order.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class PurchaseOrder(models.Model):
-    _inherit = ["purchase.order"]
+    _inherit = "purchase.order"
 
     sale_order_count = fields.Integer(
         "Number of Source Sale",
@@ -68,7 +68,7 @@ class PurchaseOrder(models.Model):
 
 
 class PurchaseOrderLine(models.Model):
-    _inherit = ['purchase.order.line']
+    _inherit = 'purchase.order.line'
 
     sale_order_id = fields.Many2one(related='sale_line_id.order_id', string="Sale Order")
     sale_line_id = fields.Many2one('sale.order.line', string="Origin Sale Item", index='btree_not_null', copy=False)

--- a/addons/sale_purchase/models/sale_order.py
+++ b/addons/sale_purchase/models/sale_order.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class SaleOrder(models.Model):
-    _inherit = ['sale.order']
+    _inherit = 'sale.order'
 
     purchase_order_count = fields.Integer(
         "Number of Purchase Order Generated",

--- a/addons/sale_purchase/models/sale_order_line.py
+++ b/addons/sale_purchase/models/sale_order_line.py
@@ -10,7 +10,7 @@ from odoo.tools.misc import get_lang
 
 
 class SaleOrderLine(models.Model):
-    _inherit = ['sale.order.line']
+    _inherit = 'sale.order.line'
 
     purchase_line_ids = fields.One2many('purchase.order.line', 'sale_line_id', string="Generated Purchase Lines", readonly=True, help="Purchase line generated by this Sales item on order confirmation, or when the quantity was increased.")
     purchase_line_count = fields.Integer("Number of generated purchase items", compute='_compute_purchase_count')
